### ISSUE TITLE
CreateBox code implementation 

### ISSUE THREAD

**Issue by Dr-Drive** [mradway/hycohanz#116]
I added and tested successfully a box drawing routine. I am sharing it and proposing it for inclusion to modeler3d.py:

```
def create_box(   oEditor, 
                    xpos, 
                    ypos, 
                    zpos, 
                    xsize, 
                    ysize,
                    zsize,
                    Name='Box1', 
                    Flags='', 
                    Color=(132, 132, 193), 
                    Transparency=0, 
                    PartCoordinateSystem='Global',
                    UDMId='',
                    MaterialValue='"vacuum"',
                    SolveInside=True,
                    IsCovered=True):
    """
    Draw a rectangle.

    Parameters
    ----------
    oEditor : pywin32 COMObject
        The HFSS editor in which the operation will be performed.
    xpos : float or hycohanz Expression object
    ypos : float or hycohanz Expression object
    zpos : float or hycohanz Expression object
        The x, y, and z coordinates of the base point of the box.
    xsize : float or hycohanz Expression object
        x-dimension of the box
    ysize : float or hycohanz Expression object
        y-dimension of the box
    ysize : float or hycohanz Expression object
        y-dimension of the box
    Name : str
        The requested name of the object.  If this is not available, HFSS 
        will assign a different name, which is returned by this function.
    Flags : str
        Flags associated with this object.  See HFSS help for details.
    Color : tuple of length=3
        RGB components of the circle
    Transparency : float between 0 and 1
        Fractional transparency.  0 is opaque and 1 is transparent.
    PartCoordinateSystem : str
        The name of the coordinate system in which the object is drawn.
    MaterialName : str
        Name of the material to assign to the object.  Name must be surrounded 
        by double quotes.
    SolveInside : bool
        Whether to mesh the interior of the object and solve for the fields 
        inside.
    IsCovered : bool
        Whether the rectangle is has a surface or has only edges.

    Returns
    -------
    str
        The actual name of the created object.

    """
    BoxParameters = [ "NAME:BoxParameters",
                    "XPosition:=", Ex(xpos).expr,
                    "YPosition:=", Ex(ypos).expr,
                    "ZPosition:=", Ex(zpos).expr,
                    "XSize:=", Ex(xsize).expr,
                    "YSize:=", Ex(ysize).expr,
                    "ZSize:=", Ex(zsize).expr]

    Attributes = [  "NAME:Attributes",
                    "Name:=", Name,
                    "Flags:=", Flags,
                    "Color:=", "({r} {g} {b})".format(r=Color[0], g=Color[1], b=Color[2]),
                    "Transparency:=", Transparency,
                    "PartCoordinateSystem:=", PartCoordinateSystem,
                    "UDMId:=", UDMId,
                    "MaterialValue:=", MaterialValue,
                    "SolveInside:=", SolveInside]

    return oEditor.CreateBox(BoxParameters, Attributes)
```

**Comment by mradway** [comment]
Thanks, I'll take a look tonight.  On the surface, I'd probably change the docstring to say "box" or "cuboid" instead of rectangle.

**Comment by mradway** [comment]
I made some minor edits to the docstring, otherwise everything's the same.  I put a short attribution note near the top, among other small edits.   Below is what it looks like.  

How do want the merge to happen, do you want me to merge it or do you want to fork it and have me pull your changes?

```
def create_box( oEditor, 
                xpos, 
                ypos, 
                zpos, 
                xsize, 
                ysize,
                zsize,
                Name='Box1', 
                Flags='', 
                Color=(132, 132, 193), 
                Transparency=0, 
                PartCoordinateSystem='Global',
                UDMId='',
                MaterialValue='"vacuum"',
                SolveInside=True,
                IsCovered=True,
                ):
    """
    Draw a 3D box.

    Note:  This function was contributed by Github user el-drive.

    Parameters
    ----------
    oEditor : pywin32 COMObject
        The HFSS editor in which the operation will be performed.
    xpos : float or hycohanz Expression object
    ypos : float or hycohanz Expression object
    zpos : float or hycohanz Expression object
        The x, y, and z coordinates of the base point of the box.
    xsize : float or hycohanz Expression object
    ysize : float or hycohanz Expression object
    zsize : float or hycohanz Expression object
        x-, y-, and z-dimensions of the box
    Name : str
        The requested name of the object.  If this is not available, HFSS 
        will assign a different name, which is returned by this function.
    Flags : str
        Flags associated with this object.  See HFSS Scripting Guide for details.
    Color : tuple of length=3
        RGB components of the circle
    Transparency : float between 0 and 1
        Fractional transparency.  0 is opaque and 1 is transparent.
    PartCoordinateSystem : str
        The name of the coordinate system in which the object is drawn.
    MaterialName : str
        Name of the material to assign to the object.  Name must be surrounded 
        by double quotes.
    SolveInside : bool
        Whether to mesh the interior of the object and solve for the fields 
        inside.
    IsCovered : bool
        Whether the rectangle is has a surface or has only edges.

    Returns
    -------
    str
        The actual name of the created object.

    """
    BoxParameters = [ "NAME:BoxParameters",
                    "XPosition:=", Ex(xpos).expr,
                    "YPosition:=", Ex(ypos).expr,
                    "ZPosition:=", Ex(zpos).expr,
                    "XSize:=", Ex(xsize).expr,
                    "YSize:=", Ex(ysize).expr,
                    "ZSize:=", Ex(zsize).expr]

    Attributes = [  "NAME:Attributes",
                    "Name:=", Name,
                    "Flags:=", Flags,
                    "Color:=", "({r} {g} {b})".format(r=Color[0], g=Color[1], b=Color[2]),
                    "Transparency:=", Transparency,
                    "PartCoordinateSystem:=", PartCoordinateSystem,
                    "UDMId:=", UDMId,
                    "MaterialValue:=", MaterialValue,
                    "SolveInside:=", SolveInside]

    return oEditor.CreateBox(BoxParameters, Attributes)    

```

**Comment by mradway** [comment]
Here's a first stab at an example function that we can include in the examples directory.  Unlike the vast majority of the examples, which haven't been updated yet, this uses the context managers to manage the HFSS COM resources:

```
import hycohanz as hfss

raw_input('Press "Enter" to connect to HFSS.>')

with hfss.App() as App:

    raw_input('Press "Enter" to create a new project.>')
    with hfss.NewProject(App.oDesktop) as P:

        raw_input('Press "Enter" to insert a new DrivenModal design named HFSSDesign1.>')
        with hfss.InsertDesign(P.oProject, "HFSSDesign1", "DrivenModal") as D:

            raw_input('Press "Enter" to set the active editor to "3D Modeler" (The default and only known correct value).>')
            with hfss.SetActiveEditor(D.oDesign) as E:

                raw_input('Press "Enter" to draw a red box named Box1.>')

                hfss.create_box(E.oEditor,  
                                1, 
                                2, 
                                3, 
                                4,
                                5,
                                Name='Box1',
                                Color=(255, 0, 0))

                raw_input('Press "Enter" to release HFSS.>')

```

**Comment by Dr-Drive** [comment]
Looks good! I am hooked on this and with the progress I have done recently, I will make sure I share more stuff regarding HFSS scripting implementations.
I like the box nomenclature because it is consistent with HFSS, apologies for the oversight.

Thank you for getting this project started! It has been super useful and powerful so far...
I noticed some aesthetic inconsistencies in the code that I could help to address in the future (in variable naming convention to be more specific). I will open more issues to keep the work organized.

**Comment by Dr-Drive** [comment]
please go ahead and incorporate it. I am not really familiar with the github environment so for the moment I will post additions/recommendations as issues. This is not optimal but code is happening really fast. Maybe I should get my hands dirty and get up to speed on github to streamline the help.

**Comment by mradway** [comment]
Git can be quite intimidating and distracting to new users, including myself, so I would advise that you consider learning Git of secondary importance for now.  Right now I think the most important thing is to get the features/corrections/improvements that are important to you into hycohanz as quickly and painlessly as possible.  Reporting them on this issue tracker is a great way to make that happen.

Thanks for the positive feedback!  I generally like scripting my HFSS models, but the HFSS scripting interface is generally too painful for me to use directly, hence hycohanz.  The HFSS scripting interface is pretty expansive, so I've generally only implemented features that are most important to me; mainly, geometry creation.  This means, of course, that new users will likely end up creating a lot of new functions, until we get more coverage of the API.  Even with the small subset of implemented features, there's still a lot there and I've made some very complex models that would have been daunting without something like hycohanz.

**Comment by mradway** [comment]
I see you've updated your profile, and you've posted some more detailed information about yourself.  Currently the attribution I have in the function reads:

```
Note:  This function was contributed by Github user el-drive.
```

How would you like that to read?  Of course, anonymous contributions are okay.

Would you also like to be added to the AUTHORS file?

**Comment by Dr-Drive** [comment]
If you think the contribution is authorship worthy, that would be an honor =).
As for names, I will follow whichever convention you have adopted so far... If you go by nicknames/github users, Dr-Drive is fine but If you go by names C.A. Donado Morcillo is works as well.

**Comment by mradway** [comment]
Merged in e07051e and 96f7c74.
